vac/raw/mix.md

```diff
@@ -0,0 +1,658 @@
+### 3. Cryptographic Primitives and Security Parameter
+
+- **Security Parameter:** $\kappa = 128$ bits provides a balance between
+  security and efficiency.
+- **Cryptographic Primitives**
+  - **Group G**: Curve25519 elliptic curve offers 128-bit security with small
+    (32-byte) group elements, efficient for both encryption and key exchange.
+  - **Hash function H**: SHA-256.
+  - **KDF:** SHA-256 (truncated to 128 bits).
+  - **AES-CTR:** AES-128 in counter mode.
+    - **Inputs:** Key `k` (16 bytes), Initialization Vector `iv` (16 bytes),
+      Plaintext `p`
+    - **Initialization Vector (IV)**: 16 bytes, chosen randomly for each
+      encryption.
+    - **Plaintext**: Data to be encrypted (_e.g.,_ routing information, message
+      payload).
+    - **Output**: Ciphertext `c` (same size as plaintext `p`).
+    - **Operation**: AES-CTR mode uses key and the counter (`iv`) to produce a
+      keystream, which is XORed with the plaintext to produce the ciphertext.
+  - **HMAC-SHA-256:** 256-bit MAC (truncated to 128 bits).
+    - **Inputs:** Key `k` (16 bytes), Message `m`
+    - **Message**: Data to be authenticated (_e.g.,_ $β$ component).
+    - **Output**: MAC `mac` (truncated to 128 bits).
+    - **Operation**: HMAC-SHA-256 uses the key and the message to produce a
+      hash-based message authentication code.
+
+### 4. Sphinx Packet Format
+
+#### 4.1 Packet Components and Sizes
+
+1. **Alpha ($α$)**: 32 bytes
+
+   - Represents a Curve25519 group element (x-coordinate in GF(2^255 - 19)).
+   - Used by mix nodes to extract shared session key using their private key.
+
+2. **Beta ($β$)**: $(2r + 1)\kappa$ bytes typically, where $r$ is the maximum
+   path length.
+
+   - Contains the encrypted routing information.
+   - We recommend a reasonable maximum path length of $r=5$, considering
+    latency/anonymity trade-offs.
+   - This gives a reasonable size of 176 bytes.
+   - We extend $β$ to accommodate next hop address and delay below.
+
+3. **Gamma ($γ$)**: $\kappa$ bytes (16 bytes)
+
+   - Output of HMAC-SHA-256, truncated to 128 bits.
+   - Ensures the integrity of the header information.
+
+4. **Delta ($δ$)**: The encrypted payload, which can be of variable size.
+   - According to the [MixMatch](https://petsymposium.org/popets/2024/popets-2024-0050.pdf)
+    paper, the Nym network uses Sphinx packets of a fixed
+    size (2413 bytes).
+   - Considering this, the maximum $δ$ size can be chosen as 2413 bytes minus
+    the header length (which will be derived below).
+
+#### 4.2 Address Format and Delay Specification
+
+In the original
+[Sphinx](https://cypherpunks.ca/~iang/pubs/Sphinx_Oakland09.pdf) paper, the
+authors use node IDs of size $\kappa$ ($16$ bytes) to represent the next hop
+addresses. To accommodate larger addresses, we'll use a combined size of
+$t\kappa$ bytes for the address and delay, where $t$ is small (_e.g.,_ $t = 2$
+or $3$).
+
+- **Delay**: 2 bytes
+  Allows delays up to 65,535 milliseconds ≈ 65 seconds.
```

Comment:
This is quite interesting, I initially thought RLN could be used for rate limiting in this context but due to the random delays that may be introduced at each intermediary node, verifying RLN proofs becomes difficult since they are strictly coupled to the sender timestamp

Reply:
(in the context of waku)